Sources/PackageCollections/Providers/GitHubPackageMetadataProvider.swift

```diff
@@ -42,6 +45,12 @@ struct GitHubPackageMetadataProvider: PackageMetadataProvider {
             return callback(.failure(Errors.invalidGitURL(reference.location)))
         }
 
+        if let cachedMetadata = self.cache?[reference] {
+            if cachedMetadata.timestamp + DispatchTimeInterval.seconds(self.configuration.cacheTTLInSeconds) > DispatchTime.now() {
```

Comment:
should we delete the entry if expired?

Reply:
We should, though if an entry has expired it also means it will be updated (unless there is failure).

I guess I am more concerned with expired entries that are never fetched again for some reason. Maybe we should limit the number of items stored in the cache and delete expired items whenever we reach the limit?

Reply:
We'll delete cache entries when we reach a limit: https://github.com/apple/swift-package-manager/pull/3368/commits/978019722b8749b0f1fbd30c4b04f9e4ff03f25d